### ISSUE TITLE
Motor Controller Features

### ISSUE THREAD

**Issue by gcjurgiel** [rambunction4330/librmb#19]
- [ ] Current limit (Both)
- [ ] Open loop power and ramp limits (Both)
- [ ] Closed loop power and ramp rates (Both)
- [ ] Test analogue and absolute encoders (SparkMax)
- [ ] Remote encoders (TalonFX)

Comment anything I forgot

**Comment by theVerySharpFlat** [comment]
Analogue encoders for the sparkmax sounds like a good feature to have, but I doubt we're going to use it next year.

**Comment by gcjurgiel** [comment]
Yeah, I think I added that before we had officially decided which motors we were using. Also I think its already implemented (with some terribly hacky pointer code), we just haven't tested it.

**Comment by gcjurgiel** [comment]
A little unrelated, but I think we might consider removing SmartMotion/MotionMagic support since I can't envision a situation we would use it. Power and ramp limits on PID outputs are more than capable of preventing jerkiness, and any motion profiling we do use is better suited to going on the roboRIO side. It just takes up space the constructor.

**Comment by theVerySharpFlat** [comment]
#31

**Comment by theVerySharpFlat** [comment]
@gcjurgiel It seems that the SparkMaxes also have support for CAN encoders. IDK if it's worth implementing though.
